src/batch/src/executor/log_row_seq_scan.rs

```diff
@@ -0,0 +1,277 @@
+    ) -> Result<BoxedExecutor> {
+        ensure!(
+            inputs.is_empty(),
+            "LogStore row sequential scan should not have input executor!"
+        );
+        let log_store_seq_scan_node = try_match_expand!(
+            source.plan_node().get_node_body().unwrap(),
+            NodeBody::LogRowSeqScan
+        )?;
+
+        let table_desc: &StorageTableDesc = log_store_seq_scan_node.get_table_desc()?;
+        let column_ids = log_store_seq_scan_node
+            .column_ids
+            .iter()
+            .copied()
+            .map(ColumnId::from)
+            .collect();
+
+        let vnodes = match &log_store_seq_scan_node.vnode_bitmap {
+            Some(vnodes) => Some(Bitmap::from(vnodes).into()),
+            // This is possible for dml. vnode_bitmap is not filled by scheduler.
+            // Or it's single distribution, e.g., distinct agg. We scan in a single executor.
+            None => Some(TableDistribution::all_vnodes()),
+        };
+        let scan_ranges = vec![ScanRange::full()];
+
+        let chunk_size = source.context.get_config().developer.chunk_size as u32;
+        let metrics = source.context().batch_metrics();
+
+        dispatch_state_store!(source.context().state_store(), state_store, {
+            let table = StorageTable::new_partial(state_store, column_ids, vnodes, table_desc);
+            Ok(Box::new(LogRowSeqScanExecutor::new(
+                table,
+                scan_ranges,
+                log_store_seq_scan_node.old_epoch.clone().unwrap(),
+                log_store_seq_scan_node.new_epoch.clone().unwrap(),
+                chunk_size as usize,
+                source.plan_node().get_identity().clone(),
+                metrics,
+            )))
+        })
+    }
+}
+impl<S: StateStore> Executor for LogRowSeqScanExecutor<S> {
+    fn schema(&self) -> &Schema {
+        &self.schema
+    }
+
+    fn identity(&self) -> &str {
+        &self.identity
+    }
+
+    fn execute(self: Box<Self>) -> BoxedDataChunkStream {
+        self.do_execute().boxed()
+    }
+}
+
+impl<S: StateStore> LogRowSeqScanExecutor<S> {
+    #[try_stream(ok = DataChunk, error = BatchError)]
+    async fn do_execute(self: Box<Self>) {
+        let Self {
+            chunk_size,
+            identity,
+            metrics,
+            table,
+            scan_ranges,
+            old_epoch,
+            new_epoch,
+            schema,
+        } = *self;
+        let table = std::sync::Arc::new(table);
+
+        // Create collector.
+        let histogram = metrics.as_ref().map(|metrics| {
+            metrics
+                .executor_metrics()
+                .row_seq_scan_next_duration
+                .with_guarded_label_values(&metrics.executor_labels(&identity))
+        });
+        // Range Scan
+        // WARN: DO NOT use `select` to execute range scans concurrently
+        //       it can consume too much memory if there're too many ranges.
+        for range in scan_ranges {
```

Comment:
It seems that `scan_ranges` is always `ScanRange::full()`? If so, we don't have to store it as a field, and just create a single scan_range here.